crates/noirc_evaluator/src/ssa/node.rs

```diff
@@ -115,6 +121,13 @@ pub enum NodeObj {
     Obj(Variable),
     Instr(Instruction),
     Const(Constant),
+    Function(FunctionKind, NodeId, ObjectType),
```

Comment:
I think we could also add the function name

Reply:
The name can be access through the FuncId, not the node_id, and this requires the SsaContext.
Having the name would allow to use if for the Display trait, instead of using the FuncId.
n.b. this is just for debugging purposes

Reply:
The name can indeed be accessed through the NodeId since you can use that to access the FuncId and thus the SsaFunction. I'll clone the name here.